src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,174 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node newNode = new Node<>(value, tail, null);
+        if (size == 0) {
+            head = newNode;
+        } else {
+            if (tail == null) {
+                tail = newNode;
+                tail.previous = head;
+                head.next = tail;
+            } else {
+                tail.next = newNode;
+                tail = newNode;
+            }
+        }
```

Comment:
code below doesn't work? If it works, I don't see any reason not to use it 
Node newNode = new Node<>(tail, value, null);
if (size == 0) {
head = newNode;
} else {
tail.next = newNode;
}
tail = newNode;
size++;

Reply:
with the proposed implementation, I receive the error "Cannot assign field "next" because "this.tail" is null."
That's why I added field initialization head.next and tail.previous.  
The corresponding code fragment is used only if there is only one head node.